### PR TITLE
CommentStore update needed for its integration in WPAndroid

### PR DESCRIPTION
**Pull request by maxme**
Major changes:
* New like / unlike actions, with tests and example usage.
* New getter `getCommentBySiteAndRemoteId`
* New way to fetch single comment and delete comment by site+remote id (needed by comments accessed by Notification).
* Fixed an issue with usage of siteId instead of selfHostedId in XMLRPC Client
* Fixed an issue with XMLRPC status string

### CODE REVIEW

File: example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_CommentTestWPCom.java [hunk 1]
```diff
@@ -142,6 +143,34 @@ public void testInstantiateAndCreateNewComment() throws InterruptedException {
...
+        // Check comment has been modified in the DB
+        CommentModel comment = mCommentStore.getCommentByLocalId(firstComment.getId());
+         assertTrue(comment.getILike());
+
+        // Unlike comment
+        mNextEvent = TEST_EVENTS.COMMENT_CHANGED;
+        mCountDownLatch = new CountDownLatch(1);
+        mDispatcher.dispatch(CommentActionBuilder.newLikeCommentAction(new RemoteLikeCommentPayload(mSite,
+                firstComment, false)));
+        assertEquals(true, mCountDownLatch.await(TestUtils.DEFAULT_TIMEOUT_MS, TimeUnit.MILLISECONDS));
```
Comment (aforcier) [inline example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_CommentTestWPCom.java]: Same as above, should be `assertTrue(...)`

File: example/src/main/java/org/wordpress/android/fluxc/example/CommentsFragment.java [hunk 2]
```diff
@@ -118,17 +129,27 @@ private void replyToFirstComment() {
         }
     }
 
+    private void likeOrUnlikeFirstComment() {
+        mDispatcher.dispatch(CommentActionBuilder.newLikeCommentAction(
+                new CommentStore.RemoteLikeCommentPayload(getFirstSite(), getFirstComment(),
+                        !getFirstComment().getILike())));
+    }
+
     @Subscribe
```
Comment (aforcier) [inline example/src/main/java/org/wordpress/android/fluxc/example/CommentsFragment.java]: Didn't notice this before, but this should be `@Subscribe(threadMode = ThreadMode.MAIN)` (same for `onCommentInstantiated()` below it, for safety).

**Comment by maxme** [comment]
@aforcier ready for round 2

**Comment by aforcier** [comment]
:shipit:

**Pull request merged**